src/manager.rs

```diff
@@ -127,34 +127,34 @@ impl<M: crate::BusMutex> BusManager<M> {
     /// my_device.do_something_on_the_bus();
     /// # }
     /// ```
-    pub fn acquire_i2c<'a>(&'a self) -> crate::I2cProxy<'a, M> {
+    pub fn acquire_i2c(&self) -> crate::I2cProxy<'_, M> {
         crate::I2cProxy { mutex: &self.mutex }
     }
 
-    /// Acquire an [`AdcProxy`] for this hardware block.
-    ///
-    /// [`AdcProxy`]: ./struct.AdcProxy.html
-    ///
-    /// The returned proxy object can then be used for accessing the bus by e.g. a driver:
-    ///
-    /// ```ignore
-    /// // For example:
-    /// // let ch0 = gpioa.pa0.into_analog(&mut gpioa.crl);
-    /// // let ch1 = gpioa.pa1.into_analog(&mut gpioa.crl);
-    /// // let adc = Adc::adc1(p.ADC1, &mut rcc.apb2, clocks);
-    ///
-    /// let adc_bus: &'static _ = shared_bus::new_cortexm!(Adc<ADC1> = adc).unwrap();
-    /// let mut proxy1 = adc_bus.acquire_adc();
-    /// let mut proxy2 = adc_bus.acquire_adc();
-    ///
-    /// proxy1.read(ch0).unwrap();
-    /// proxy2.read(ch1).unwrap();
-    ///
-    /// ```
+    // /// Acquire an [`AdcProxy`] for this hardware block.
+    // ///
+    // /// [`AdcProxy`]: ./struct.AdcProxy.html
+    // ///
+    // /// The returned proxy object can then be used for accessing the bus by e.g. a driver:
+    // ///
+    // /// ```ignore
+    // /// // For example:
+    // /// // let ch0 = gpioa.pa0.into_analog(&mut gpioa.crl);
+    // /// // let ch1 = gpioa.pa1.into_analog(&mut gpioa.crl);
+    // /// // let adc = Adc::adc1(p.ADC1, &mut rcc.apb2, clocks);
+    // ///
+    // /// let adc_bus: &'static _ = shared_bus::new_cortexm!(Adc<ADC1> = adc).unwrap();
+    // /// let mut proxy1 = adc_bus.acquire_adc();
+    // /// let mut proxy2 = adc_bus.acquire_adc();
+    // ///
+    // /// proxy1.read(ch0).unwrap();
+    // /// proxy2.read(ch1).unwrap();
+    // ///
+    // /// ```
 
-    pub fn acquire_adc<'a>(&'a self) -> crate::AdcProxy<'a, M> {
-        crate::AdcProxy { mutex: &self.mutex }
-    }
+    // pub fn acquire_adc<'a>(&'a self) -> crate::AdcProxy<'a, M> {
```

Comment:
Could you help me understand why this section of the code is commented out, especially since the corresponding test was also removed? I’m curious whether this change addresses a particular issue or if it’s related to clean code practices.

Reply:
As I wrote this code a while ago, I don't fully remember - however I believe it is because there is no implementation of ADC or analogue pins in embedded-hal v1.0.0, whereas there was before. This code can be removed rather than commented out, I just commented it as this may be introduced in a later embedded-hal version (e.g. v1.1.0)